### ISSUE TITLE
K.function not works in multigpu mode

### ISSUE THREAD

**Issue by iperov** [keras-team/keras#10365]
```
m1 = keras.utils.multi_gpu_model(m1, ...)
output_tensor = m1( input_tensor )
func = K.function ([input_tensor], [output_tensor])
func(...)
```
not works in multigpu mode

**Comment by iperov** [comment]
keras not flexible while using only standard Model() 
for example I checking two pictures via multiply on mask, also model training this mask, so I have to pass mask twice. With K.function I avoiding this problem, but it not works on multiple gpu :(
